tools/ddprobe.cpp

```diff
@@ -69,8 +71,128 @@ syncThreadDesktop() {
   CloseDesktop(hDesk);
 }
 
+/**
+  * @brief Determines if a given frame is valid by checking if it contains any non-dark pixels.
+  *
+  * This function analyzes the provided frame to determine if it contains any pixels that exceed a specified darkness threshold.
+  * It iterates over all pixels in the frame, comparing each pixel's RGB values to the defined darkness threshold.
+  * If any pixel's RGB values exceed this threshold, the function concludes that the frame is valid (i.e., not entirely dark) and returns `true`.
+  * If all pixels are below or equal to the threshold, indicating a completely dark frame, the function returns `false`.
+
+  * @param mappedResource A reference to a `D3D11_MAPPED_SUBRESOURCE` structure containing the mapped subresource data of the frame to be analyzed.
+  * @param frameDesc A reference to a `D3D11_TEXTURE2D_DESC` structure describing the texture properties, including width and height.
+  * @param darknessThreshold A floating-point value representing the threshold above which a pixel's RGB values are considered dark. The value ranges from 0.0f to 1.0f, with a default value of 0.1f.
+  * @return bool Returns `true` if the frame contains any non-dark pixels, indicating it is valid; otherwise, returns `false`.
```

Comment:
```suggestion
  * @return Returns `true` if the frame contains any non-dark pixels, indicating it is valid; otherwise, returns `false`.
```

I forgot this in my last review. We don't specify a return type. It's automatic in doxygen and the IDE.

Applies to both cases.